File: Algorithms/Greedy/UniqueSummands.java

import java.util.*;

public class UniqueSummands {

    public static List<Integer> getUniqueSummands(int num) {

        List<Integer> summands = new ArrayList<>();

        if (num <= 2) {

            summands.add(num);

        } else {

            int k = 1;

            while (num > 2 * k) {

                num -= k;

                summands.add(k++);

            }

            summands.add(num);

        }

        return summands;

    }

    public static void main(String[] args) {

        for (int n : getUniqueSummands(8)) {

            System.out.println(n);

        }

    }

}

File: Algorithms/Greedy/UniqueSummands.js
"use strict";

let getUniqueSummands = (num) => {

    let summands = [];

    if (num <= 2) {

        summands.push(num);

    } else {

        let k = 1;

        while (num > 2 * k) {

            num -= k;

            summands.push(k);

            k += 1;

        }

        summands.push(num);

    }

    return summands;

}

for (let n of getUniqueSummands(8)) {

    console.log(n);

}
